### ISSUE TITLE
Graphics overhaul tracking

### ISSUE THREAD

**Issue by dgw** [Warehousing/Warehousing#51]
This ticket began as a simple "re-export the sprites in higher resolution" idea, and has evolved over nearly 9 months to become quite complicated. Here's a reasonably concise list of what we actually want to do, based on changes from Factorio 0.17 so far (as of 0.17.29) and plans revealed in recent FFF posts:

  - [x] Redesign buildings to better fit new HD style used by Wube (especially chests, per [FFF 283](https://factorio.com/blog/post/fff-283)) [#88]
    - [x] Add robot pick-up/drop-off animations to the logistic warehouses/storehouses (also [FFF 283](https://factorio.com/blog/post/fff-283)) [#88]
      - Factorio [0.17.5 added](https://forums.factorio.com/66705) a `LogisticContainerPrototype::landing_location_offset` property to control where robots dock, upon [our request](https://forums.factorio.com/65702), which means our new buildings won't be constrained to putting the robot docking hatch in the center
  - [x] Overhaul icons to match [#88]
    - [x] High resolution item icons
      - [x] At least bump to 64x64 per [FFF 277](https://factorio.com/blog/post/fff-277) [#88]
      - [x] Multiple icon resolutions with simplified designs at smaller sizes [#88]
        - Factorio itself will likely use a mix of two approaches, based on [FFF 290](https://factorio.com/blog/post/fff-290) and [FFF 291](https://factorio.com/blog/post/fff-291)
        - We should simply use whichever gives us the best results for _our_ icons
    - [x] Technology icons to match base game techs as of Factorio 1.1: [#88]
      - "Base game technology icons now use 4 mipmaps and icon size 256 instead of icon size 128 and no mipmaps"
      - Moved here from #82, as despite being posted in [the "breaking changes" forum thread](https://forums.factorio.com/viewtopic.php?p=521904#p521904) it doesn't break us
    - [x] Make it easier to tell which entity icon is which at-a-glance (per [mod portal feedback](https://mods.factorio.com/mod/Warehousing/discussion/5c32902cb10da7000d0a8070) indicating that the current icon sizes are easily confused) [#88]
      - ~~_Especially_ important if we decide to integrate "Small" and "Tiny" warehouses as proposed by @JDOGG88 in #68—there would then be at least six different icons to distinguish~~
  - [ ] ~~Apply look-up tables (LUTs) to sprites/icons? ([FFF 320](https://factorio.com/blog/post/fff-320))~~
    - NAK; #88's new graphics were created long after the base game color tweaks, and should just match without needing to apply any LUTs
  - [x] Make `thumbnail.png` for the mod portal/GUI [#88]
    - Technically already done in dbf6e7c, but the current demo map/screenshot will need to be tweaked/regenerated after the rest of this list is finished

_Original issue description from mid-2018 follows._

----

It would be nice to launch high-resolution sprites along with, or shortly after, Factorio 0.17's release. Warehousing will need to be updated anyway, and that update might as well come with a nice bonus. (Wube will no doubt upgrade a number of base-game entities to HD, as they have in past major updates, so Warehousing would really just be joining the party. :smile_cat:)

@Nova-Kast I'm assigning this issue to you because you're the only one who can actually generate new images. (If you want to do #32 properly at the same time, that would be fine too. :grin:)

**Comment by dgw** [comment]
[FFF 269](https://factorio.com/blog/post/fff-269) today announced Factorio 0.17 is planned to release in January 2019. That's a good target for getting this done.

**Comment by dgw** [comment]
User feedback regarding icons (which will likely be redone along with the sprites):

> Because the icon on the crafting bar for the Storehouse is larger then the Warehouse, I often click on the wrong one. My Suggestion would be to alter the icon image slightly to have the Warehouse visually larger then the Warehouse ... that's basically it.

— https://mods.factorio.com/mod/Warehousing/discussion/5c32902cb10da7000d0a8070

**Comment by dgw** [comment]
[FFF 277](https://factorio.com/blog/post/fff-277) has buried in it a little graphical tidbit: Improved GUI scaling support will bring along higher-resolution icons. They will be upgrading item icons from 32x32 pixels to 64x64, to support larger GUI scales without making the icons look bad.

Warehousing should go along with this when the graphics update is done.

**Comment by dgw** [comment]
GitHub, what do you mean I unassigned Nova…? Must have had something to do with repo permissions and/or moving the repo into an organization. GitHub messes with stuff when you do that. :slightly_frowning_face:

Well anyway… [FFF 283](https://factorio.com/blog/post/fff-283) announced a couple graphics-related things that we probably want to act on:

  1. The in-game mod GUI will show a thumbnail image for each mod, and the way it's implemented requires including a `thumbnail.png` file (144x144 pixels) in the mod package.
  2. Chests were upgraded to high-resolution. The most relevant change is that logistic chests now have a roboport-like animation—something it would be nice for Warehousing's buildings to have, too.

I'll see about picking a `thumbnail.png` to include in future releases—probably just the one we have now, at first. Whatever we use, an updated thumbnail should come with the graphics refresh, whenever it actually happens (so the thumbnail will match the new entity sprites).

Adding animations to the warehouse/storehouse buildings will _probably_ be its own project, separate from upgrading the sprites to high-res. That is, unless @Nova-Kast is feeling really ambitious and wants to do everything at once. :grin:

**Comment by dgw** [comment]
Today's [FFF 290](https://factorio.com/blog/post/fff-290) revealed that Wube are working toward better icon scaling. It's possible Factorio will gain support for multiple icon sizes to be used at different zoom levels, and we should keep this on the radar until they come to a final decision on how they want to handle it.

I'll update the OP with a simple list of things we're keeping tabs on, because this issue has gotten quite complicated with all the FFF-based follow-ups.

**Comment by JDOGG88** [comment]
Any idea on a potential timeline for the new graphics or is the plan to wait until Wube makes their final decision?  I'm just anxious to see what the new graphics are going to look like. 😂

**Comment by dgw** [comment]
> Any idea on a potential timeline for the new graphics

When @Nova-Kast is ready and able to work on them, they'll get done. No timeline at present.

**Comment by dgw** [comment]
Tweaked icon-size list entry (and children) to include [FFF 291](https://factorio.com/blog/post/fff-291).

**Comment by dgw** [comment]
Added LUTs to the list (see [FFF 320](https://factorio.com/blog/post/fff-320)).

**Comment by JDOGG88** [comment]
So I was thinking about making a "plugin" for small and tiny warehousing just until the new graphics are done and/or depending on the popularity of it.  It would just be a separate mod that makes Warehousing dependent and adds in the small and tiny versions.  It may make it easier to decide whether or not to add them into the main mod based on popularity.  I'd like your feedback before I create it and upload it to the mod portal.

**Comment by dgw** [comment]
@JDOGG88 An add-on mod is fine with me. If you'd be creating and distributing derivative works based on this mod's graphics, though, you should ask @Nova-Kast first. (Quickest way to do that is through @Anoyomouse; PM on the Factorio forums if neither of them replies here in a day or two.)

**Comment by JDOGG88** [comment]
Nah it wouldn't be anything derivative using the graphics.  It would just be pulling my Small & Tiny code out and use it to make a "Warehousing Compression" mod with the ultimate goal of seeing how people like it and whether or not to actually add it into the main mod.

**Comment by dgw** [comment]
Might as well have at it, @JDOGG88. Have fun!

**Comment by JDOGG88** [comment]
If you have the time and want to take a look [https://github.com/JDOGG88/Warehouse-Compression](url)

**Comment by dgw** [comment]
Hey @Nova-Kast, are you still working on these new graphics?

**Comment by Anoyomouse** [comment]
Linking to PR #88, snouz has done some amazing work redoing all the sprites as well as tweaking them for in game

**Comment by dgw** [comment]
Once #88 is done I'll try to figure out if LUTs are still relevant, or if this whole thing can be closed out.

**Comment by snouz** [comment]
> Once #88 is done I'll try to figure out if LUTs are still relevant, or if this whole thing can be closed out.

Here's Wube LUT script
https://github.com/wube/factorio-LUT-tool
I have no idea if it's relevant. It says the script "applies colors" to an image, but there aren't several images for each entity, so maybe that's just something that is done in-engine?

**Comment by dgw** [comment]
I don't think the LUT script is relevant. In fact, I think that whole checklist item can just… not happen. We don't need it, since you specifically offered to create new graphics because you could make them match Factorio's current look.

So, in anticipation of the next release with those new graphics that check all the boxes above, I'll just close this out and focus on testing/releasing when I get the time!
